src/test/java/org/pesmypetcare/webservice/dao/MedicationDaoTest.java

```diff
@@ -0,0 +1,461 @@
+    private static String owner;
+    private static String petName;
+    private static String date;
+    private static String date2;
+    private static String name;
+    private static String dateName;
+    private static String field;
+    private static int value;
+
+
+    private static final String USERS_KEY = "users";
+    private static final String PETS_KEY = "pets";
+    private static final String MEDICATION_KEY = "medications";
+    private final String EXCECUTION_EXC_MSG = "Should throw DatabaseAccessException when ExecutionException received";
+    private final String INTERRUPTED_EXC_MSG = "Should throw DatabaseAccessException when InterruptedException "
+            + "received";
+    private final String DOCUMENT_NOT_EXISTS_EXC_MSG = "Should throw DatabaseAccessException when med document"
+            + " doesn't exist";
+
+    @Mock
+    private Firestore db;
+    @Mock
+    private CollectionReference usersRef;
+    @Mock
+    private DocumentReference ownerRef;
+    @Mock
+    private CollectionReference petsRef;
+    @Mock
+    private DocumentReference petRef;
+    @Mock
+    private CollectionReference medsRef;
+    @Mock
+    private DocumentReference medRef;
+    @Mock
+    private ApiFuture<QuerySnapshot> futureQuery;
+    @Mock
+    private QuerySnapshot querySnapshot;
+    @Mock
+    private ApiFuture<DocumentSnapshot> futureDocument;
+    @Mock
+    private DocumentSnapshot documentSnapshot;
+    @Mock
+    private List<QueryDocumentSnapshot> medsDocuments;
+    @Mock
+    private Iterator<QueryDocumentSnapshot> it;
+
+    @InjectMocks
+    private MedicationDao medDao = new MedicationDaoImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        medList = new ArrayList<>();
+        medEntity = new MedicationEntity();
+        owner = "usuario_de_pruebas";
+        petName = "linux";
+        date = "26-03-2020";
+        date2 = "26-04-2020";
+        dateName = "26-03-2020%Cloroform";
+        field = "duration";
+        value = 4;
+        name = "Cloroform";
+    }
+
+    @Test
+    public void shouldCreateMedOnDatabaseWhenRequested() {
+        given(db.collection(anyString())).willReturn(usersRef);
+        given(usersRef.document(anyString())).willReturn(ownerRef);
+        given(ownerRef.collection(anyString())).willReturn(petsRef);
+        given(petsRef.document(anyString())).willReturn(petRef);
+        given(petRef.collection(anyString())).willReturn(medsRef);
+        given(medsRef.document(anyString())).willReturn(medRef);
+        given(medRef.set(isA(MedicationEntity.class))).willReturn(null);
+
+        medDao.createMedication(owner, petName, dateName, medEntity);
+
+        verify(db).collection(same(USERS_KEY));
+        verify(usersRef).document(same(owner));
+        verify(ownerRef).collection(same(PETS_KEY));
+        verify(petsRef).document(same(petName));
+        verify(petRef).collection(same(MEDICATION_KEY));
+        verify(medsRef).document(same (dateName));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['(' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)